src/util.ts

```diff
@@ -11,11 +11,13 @@ export function mergeHeaders(...allHeaders: (Headers | object)[]): Headers {
   for (let ithHeader of allHeaders) {
     if (ithHeader instanceof Headers) {
       for (let [k, v] of ithHeader) {
-        result.set(k, v)
+        ithHeader.get('Set-Cookie') !== null ? result.append(k, v) : result.set(k, v)
```

Comment:
apologies for no linter on this repo - Can make this a regular ol' multi-line if statement rather than a single line ternary?